subprojects/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFileVisitDetails.java

```diff
@@ -26,45 +26,16 @@
 
 public class DefaultFileVisitDetails extends DefaultFileTreeElement implements FileVisitDetails {
     private final AtomicBoolean stop;
-    private final boolean isDirectory;
-    private final long size;
-    private final long lastModified;
 
     public DefaultFileVisitDetails(File file, RelativePath relativePath, AtomicBoolean stop, Chmod chmod, Stat stat) {
-        this(file, relativePath, stop, chmod, stat, !relativePath.isFile());
-    }
-
-    public DefaultFileVisitDetails(File file, RelativePath relativePath, AtomicBoolean stop, Chmod chmod, Stat stat, boolean isDirectory) {
-        this(file, relativePath, stop, chmod, stat, isDirectory, file.lastModified(), file.length());
-    }
-
-    public DefaultFileVisitDetails(File file, RelativePath relativePath, AtomicBoolean stop, Chmod chmod, Stat stat, boolean isDirectory, long lastModified, long size) {
         super(file, relativePath, chmod, stat);
         this.stop = stop;
-        this.isDirectory = isDirectory;
-        this.lastModified = lastModified;
-        this.size = size;
     }
 
     public DefaultFileVisitDetails(File file, Chmod chmod, Stat stat) {
         this(file, new RelativePath(!file.isDirectory(), file.getName()), new AtomicBoolean(), chmod, stat);
     }
 
-    @Override
-    public boolean isDirectory() {
```

Comment:
I think we should keep remembering if the file was a directory or not. Or use the information from `relativePath`.

Reply:
👍 Fixing.